core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/Parameter.kt

```diff
@@ -19,30 +19,31 @@ object ParameterManager {
 
 sealed class Parameter<T : ValidType>(
     private val value: Any,
-    private val parameterName: String? = null,
+    private val parameterName: String?,
 ) : TypeExpression<T> {
-    override fun toDopeQuery(): DopeQuery = when (parameterName) {
+    override fun toDopeQuery() = when (parameterName) {
         null -> {
             val unnamedParameterCount = "\$${ParameterManager.count}"
             DopeQuery(
                 queryString = unnamedParameterCount,
                 parameters = mapOf(unnamedParameterCount to value),
             )
         }
+
         else -> DopeQuery(
             queryString = "\$$parameterName",
             parameters = mapOf(parameterName to value),
         )
     }
 }
 
-class NumberParameter(value: Number, parameterName: String?) : Parameter<NumberType>(value, parameterName)
+class NumberParameter(value: Number, parameterName: String? = null) : Parameter<NumberType>(value, parameterName)
```

Comment:
Why are we setting a default parameter here?

Reply:
We previously had the default value in the sealed class `Parameter` and not in these Sub classes, but that doesn't make much sense, because you can't create an instance of a Sealed class, so the User would only use The Subclasses. And because the user shouldn't be needed to pass `null` I added these Default values.